src/utils.py

```diff
@@ -28,49 +28,41 @@ def risk_score(los):
         return 1
 
 
-def train_model(gsc, X_train, y_train, evaluator):
-    """Uses a GridSearchCV instance to find a reasonable model, and store
-    performance and fitted model into a python dict
+def train_and_test_model(estimator, X_train, y_train, X_test, y_test, scoring_metric):
+    """Trains and tests a model, returning results in a python dict
 
     Parameters:
 
-        gsc (sklearn.model_selection.GridSearchCV object): defined model
+        estimator (object): estimator object implementing 'fit'
         X_train (pandas dataframe): training dataframe with features
         y_train (pandas dataframe): training dataframe with targets
-        evaluator (object): one of "mean_absolute_error" or "f1_weighted"
+        X_test (pandas dataframe): test dataframe with features
+        y_test (pandas dataframe): test dataframe with targets
+        scoring_metric (object): one of "mean_squared_error" or "f1_weighted"
+
 
     Returns:
 
         (dict): resulting fitted model and performance metrics
     """
 
-    grid_result = gsc.fit(X_train, y_train)
-
-    # note model scoring approach defined in gsc
-    model = {
-        "cv_metric_mean": np.round(
-            grid_result.cv_results_["mean_test_score"][grid_result.best_index_], 3
-        ),
-        "cv_metric_std": np.round(
-            grid_result.cv_results_["std_test_score"][grid_result.best_index_], 2
-        ),
-        "model": grid_result.best_estimator_,
-    }
-
-    # retrain the best estimator on the full training set - note that refit=True
-    # does not appear to do this
-    model["model"].fit(X_train, y_train)
-
-    # generate predictions, and make sure we remove any negative length of stay
-    # predictions
-    preds = np.clip(model["model"].predict(X_train), 0, None)
-
-    # calculate final performance
-    if evaluator == "mean_absolute_error":
-        model["mae"] = np.round(mean_absolute_error(y_train, preds), 3)
-    elif evaluator == "f1_weighted":
-        model["f1_weighted"] = np.round(f1_score(y_train, preds, average="weighted"), 3)
+    model = {}
+
+    # fit the model based on the training data
+    model["model"] = estimator.fit(X_train, y_train)
+
+    # generate predictions of the training set
+    preds_train = np.clip(model["model"].predict(X_train), 0, None)
+    preds_test = np.clip(model["model"].predict(X_test), 0, None)
+
+    # calculate performance
+    if scoring_metric == "mean_squared_error":
```

Comment:
Could be useful to save the scoring metric into the model dictionary too, so that when the user picks up a set of models and prints the result they know the metric saved into the model when it was created..